src/features/voyage-planner.tsx

```diff
@@ -1,30 +1,174 @@
+import React, {useState, useEffect} from 'react'
 import RouteMap from "./route-map"
+import { useAppDispatch, useAppSelector } from './../services/hooks'
+import { Port, PortsState} from './../services/ports-reducer'
+import VoyagePort from './../features/voyage-port'
+import axios, { AxiosError } from "axios"
 import "./voyage-planner.css"
 
 interface VoyagePlannerParams{
+    selectedPorts?: Port[]
+}
+
+const initialState: VoyagePlannerParams = {
+    selectedPorts: []
+}
 
+const initialPortState: PortsState = {
+    loading: true,
+    error: '',
+    count: 0,
+    offset: 0,
+    ports: []
 }
 
+export const LOAD_USERS_LOADING = 'REDUX_THUNK_LOAD_USERS_LOADING';
+export const LOAD_USERS_SUCCESS = 'REDUX_THUNK_LOAD_USERS_SUCCESS';
+export const LOAD_USERS_ERROR = 'REDUX_THUNK_LOAD_USERS_ERROR';
+export const LOAD_USERS_SUCCESS_REC = 'REDUX_THUNK_LOAD_USERS_SUCCESS_REC';
+export const ADD_VOYAGE_PORTS = "REDUX_THUNK_ADD_VOYAGE_PORTS"
+
 function VoyagePlanner(params: VoyagePlannerParams){
+    const ports = useAppSelector(state => state.ports);
+    const voyage = useAppSelector(state => state.voyage);
+    const [currentSelectedPort, setCurrentSelectedPort] = useState("")
+    const [addedPorts, setAddedPorts] = useState([] as Port[])
+    const dispatch = useAppDispatch();
+    
+    const changeSelectedPort = (selectedPort: string) => {
+        var selectBox = document.getElementById("port-select") as HTMLSelectElement;
+        var selectedValue = selectBox.options[selectBox.selectedIndex].value;
+
+        setCurrentSelectedPort(selectedValue);
+    }
+    
+    const addPortToSelectedPorts = () => {
+        if (currentSelectedPort == "") {
+            setCurrentSelectedPort(ports.ports[0].name);
+            setAddedPorts(prevArray => [...prevArray, ports.ports[0]])
+            dispatch( {type: ADD_VOYAGE_PORTS, data:ports.ports[0]})
+        }
+
+        for (let index = 0; index < ports.ports.length; index++) {
+            const element = ports.ports[index] as Port;
+            if (element.name == currentSelectedPort && !containsObject(addedPorts, element)) {
+                setAddedPorts(prevArray => [...prevArray, element])
+                //let updatedList: Port[] = addedPorts.concat(element);
+                dispatch( {type: ADD_VOYAGE_PORTS, data: element})
+            }
+        }
+    }
+
+    const containsObject = (array: Port[], element: Port): boolean => {
+        for (let i = 0; i < array.length; i++) {
+            if (array[i].name === element.name) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    useEffect(() => {
+        dispatch({type: LOAD_USERS_LOADING});
+
+        let results: PortsState = initialPortState;
+        let baseURL = "https://8u9tblsay0.execute-api.us-east-1.amazonaws.com/default/zn-frontend-challenge-port-service"
+        let headers = { 'Access-Control-Allow-Origin': '*', 'Content-Type' : 'application/json;charset=utf-8'}
```

Comment:
This header breaks CORS - it can not be included in the request.  The CORS issue was fixed prior to your submission, but this header meant that you were not able to call the API.  For future reference, you can also get around CORS issues by using the built-in create-react-app proxy functionality.